### ISSUE TITLE
iOS example app does not seem to work

### ISSUE THREAD

**Issue by basvdijk** [KasemJaffer/receive_sharing_intent#296]
When I run the app from the example folder on https://github.com/KasemJaffer/receive_sharing_intent/tree/master/example on iOS it does not seem to work. 

When sharing an image:

- You see the app in the list
![image](https://github.com/KasemJaffer/receive_sharing_intent/assets/644550/3b0a61a3-6388-481f-bd17-744ffadc2748)
- Then for a split second you see this window pop up
![image](https://github.com/KasemJaffer/receive_sharing_intent/assets/644550/d7e97005-4872-4b92-a1c3-3f396f51acee)
- Then nothing happens

I tested this on an iPhone 15 Pro Max emulator running iOS 17.4

**Comment by wanwanland** [comment]
me to 😢
I tested this on an iPhone 14 Pro actual machine

**Comment by zhponng** [comment]
me to 
Any new developments?

**Comment by olerhan** [comment]
me to 😢
I checked again and again to see what I might have done wrong. Has anyone found a solution?

**Comment by basvdijk** [comment]
My biggest concern at the moment is reliability of this feature / package. It has been 2 weeks without reply to this issue. I am holding back to use this package for now. If I use this in production and it breaks, I do not want to risk to have a broken feature for weeks.

**Comment by olerhan** [comment]
I managed to get it to work. While the package publisher could have provided clearer instructions, it's important to understand some key points when adding packages to your application. After trying many forked versions of this package and examining the package codes, I've gained full experience with it. The issue is actually quite simple. Here are the key steps:

Pay attention to the Info.plist files. Insert the required codes correctly.
Provide the host and extension IDs in the correct format.
After filling in the ShareViewController content, be sure to run pod install in the iOS directory before running the application.
Follow the cycle described by the publisher carefully.
If you are not using custom IDs, fill them in manually in the Info.plist table.
The original package works fine and is more compact than other packages. It also has a supportive community.

I hope this helps others who are facing similar issues.

**Comment by tamara-jovanoska** [comment]
@olerhan did you make it work for the newest version or versions prior 1.6.0 ?

**Comment by tamara-jovanoska** [comment]
For me what made it work is filling the custom IDs manually for the Runner and Share Extension under `Build Settings`, what @olerhan suggested above.

<img width="855" alt="Screenshot 2024-06-03 at 16 31 18" src="https://github.com/KasemJaffer/receive_sharing_intent/assets/98604180/31259a36-5279-4567-98a0-455bb49fed1b">

**Comment by tamara-jovanoska** [comment]
for me what made it work is adding the custom IDs manually for both Runner and Sharing Extension under Build Settings, what @olerhan suggested above.
<img width="855" alt="Screenshot 2024-06-03 at 16 31 18" src="https://github.com/KasemJaffer/receive_sharing_intent/assets/98604180/54924750-b268-4618-b9bb-96ee4c833226">

**Comment by Guscccc** [comment]
Same here. I cloned the example and it doesn't work for iOS either. The app doesn't receive the shared files. Confirmed on iOS 17.5 simulator.

**Comment by bigzhu** [comment]
I think the issue is not caused by the configuration, but by this: 
* #294

**Comment by olerhan** [comment]
For me everything still works fine. Everyone just needs to be more careful about using the package. This issue (296 or 294) is not a real issue…

**Comment by strohercocari** [comment]
> For me everything still works fine. Everyone just needs to be more careful about using the package. This issue (296 or 294) is not a real issue…

hey, can u provide the ShareViewController.swift?

**Comment by KhuntJenish** [comment]
> I managed to get it to work. While the package publisher could have provided clearer instructions, it's important to understand some key points when adding packages to your application. After trying many forked versions of this package and examining the package codes, I've gained full experience with it. The issue is actually quite simple. Here are the key steps:
> 
> Pay attention to the Info.plist files. Insert the required codes correctly. Provide the host and extension IDs in the correct format. After filling in the ShareViewController content, be sure to run pod install in the iOS directory before running the application. Follow the cycle described by the publisher carefully. If you are not using custom IDs, fill them in manually in the Info.plist table. The original package works fine and is more compact than other packages. It also has a supportive community.
> 
> I hope this helps others who are facing similar issues.

@olerhan You’re absolutely right — I’ve been facing image sharing intent issues for several days as well.

In my case, I’m using multiple **flavors (development & production)** in my Flutter app, and most of the problems were due to **native misconfigurations**. I spent almost **2–3 weeks debugging**, but I finally figured out the full setup:

### What caused the issue

* Share Extension and App Group were not configured correctly **per flavor**.
* `CUSTOM_GROUP_ID` was not properly assigned for each flavor in **Build Settings**.
* In Xcode → *Runner → Info → Configurations*, I had selected the **wrong flavor** for the execution target.

### Final Fix

* Created **separate Share Extension bundle identifier  + App Group IDs** for each flavor.
* Ensured `Info.plist` values were filled **explicitly** for each build.
* Set `CUSTOM_GROUP_ID` correctly in Build Phases for both Development and Production.
* Re-ran `pod install` every time after making changes.

✅ Now image sharing is working **properly on both Android and iOS**.

---

### One Remaining Question

Right now, when I share an image (from Gallery or using a quick screenshot on iOS), it **shows an intermediate popup** before redirecting to my app.

However, I noticed that some apps (like **Discord**) **redirect directly into their app without showing that popup UI**.

> Is there a way to achieve **instant redirect bypassing the popup UI entirely** in iOS Share Extensions?

Would appreciate any guidance if you have insights on this approach.

---
this is my ShareViewController.swift code
```
import receive_sharing_intent

class ShareViewController: RSIShareViewController {
    
    // Automatically redirect to main app without showing any UI
    override func shouldAutoRedirect() -> Bool {
        return true  // Enable automatic redirect
    }
    
}
```
---
this is my current working app video

https://github.com/user-attachments/assets/c487f3e5-2106-4519-b398-3efb6da8f625

https://github.com/user-attachments/assets/1ea247d3-e660-41fb-820f-81879f118a81

---

### Additional Observation

When using the iOS Share Extension to share an image from a freshly taken screenshot, the app does not display the selected image. Instead, only the keyboard opens with no preview. The shared media is not passed correctly into the app on real devices.

This happens because receive_sharing_intent uses a fallback mechanism — it creates a temporary image file (TempImage.png) inside the App Group container when iOS provides only an in-memory UIImage (not a file URL).
However, on real devices, the shared file is not accessible to the main app due to App Group misconfiguration or sandboxing, even though it works fine on the simulator.

---

### 🔗 Related Discussions (React Native)

React Native discussion — similar issue solved natively
👉 https://github.com/achorein/expo-share-intent-demo/issues/28
👉 https://github.com/ajith-ab/react-native-receive-sharing-intent/issues/56

If anyone has managed to achieve instant redirect without showing the intermediate share popup in Flutter (not React Native), please share your insights here
